actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -822,8 +822,10 @@ def url_for(options, route_name = nil, url_strategy = UNKNOWN, method_name = nil
         path = route_with_params.path(method_name)
         params = route_with_params.params
 
-        if options.key? :params
+        if options[:params].is_a?(Hash)
```

Comment:
So this broke our app quite extensively as it's not rare for `options[:params]` to be an `ActionController::Parameters`. 

I'll fix it in https://github.com/rails/rails/pull/42603, feel free to chime in.